### PR TITLE
SPARKC-446 Enabling PerPartitionLimits for CassandraTable and JoinWithCassandraTable

### PR DESCRIPTION
**Pull request by RussellSpitzer**
c2f1cdb (Russell Spitzer, 6 minutes ago)
    SPARKC-446: Enable PerPartitionLimit as a CassandraRDD Function

    Cassandra 3.6 enables PerPartitionLimit. To accesses this the underlying
    CassandraRDD limit field now holds a `CassandraLimit` opition. There are
    two possible values at the moment.

    SparkPartitionLimit - Which limits the number of rows returned per Spark
    Partition CassandraPartitionLimit - Which limits the number of rows
    returned per Cassandra Partition

    They are accessed through the same api as previously existed

    CassandraRDD.limit() - Sets the SparkPartitionLimit like previously
    CassandraRDD.perPartitionLimit - Sets the CassandraPartitionLimit

ce8ea46 (Russell Spitzer, 6 minutes ago)
    SPARKC-446: Change Cassandra Version to 3.6

    Only 3.6 Supports Per Partition Limit

### CODE REVIEW

File: spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraLimit.scala [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package com.datastax.spark.connector.rdd
+
+sealed trait CassandraLimit
+
+case class CassandraPartitionLimit(perPartitionLimit: Long) extends CassandraLimit {
```
Comment (jacek-lewandowski) [inline spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraLimit.scala]: I'd name the parameter `rowsNumber`

File: spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraLimit.scala [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  require(perPartitionLimit > 0, s"$perPartitionLimit <= 0. Per Partition Limits must be greater than 0")
+}
+case class SparkPartitionLimit(limit: Long) extends CassandraLimit {
```
Comment (jacek-lewandowski) [inline spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraLimit.scala]: I'd name the parameter `rowsNumber`

File: spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraLimit.scala (hunk 1 above)
Comment (RussellSpitzer) [inline spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraLimit.scala]: done!

File: spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraLimit.scala (hunk 2 above)
Comment (RussellSpitzer) [inline spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraLimit.scala]: and done!

**Pull request merged**